bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1658,31 +1658,30 @@ impl<'a> CircuitInputStateRef<'a> {
     ) -> Result<(CopyEventSteps, CopyEventPrevBytes), Error> {
         let src_addr = src_addr.into().0;
         let dst_addr = dst_addr.into().0;
-        let src_addr_end = src_addr_end.into().0;
         let bytes_left = bytes_left.into().0;
         let src_copy_end = src_addr + bytes_left;
         let dst_copy_end = dst_addr + bytes_left;
+        let src_addr_end = src_copy_end.min(src_addr_end.into().0);
 
         if bytes_left == 0 {
             return Ok((vec![], vec![]));
         }
 
-        // Extend call memory.
-        let memory = &mut self.call_ctx_mut()?.memory;
-        memory.extend_for_range(dst_addr.into(), bytes_left.into());
```

Comment:
`memory.extend_for_range(dst_addr.into(), bytes_left.into());`

This removed variant was the most correct, because it implements pure EVM semantics of memory expansion, while "slot" and "full_length" are our own internal invention. Although this happens to give the same result.